ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController.swift

```diff
@@ -2657,11 +2657,12 @@ extension BrowserViewController: TabDelegate {
       tab.requestBlockingContentHelper,
     ]
 
-    let braveTranslateScriptHandler = BraveTranslateScriptHandler(tab: tab, delegate: self)
+    let braveTranslateTabHelper = BraveTranslateTabHelper(tab: tab, delegate: self)
+
     injectedScripts.append(
-      BraveTranslateScriptLanguageDetectionHandler(tab: tab, delegate: braveTranslateScriptHandler)
+      BraveTranslateScriptLanguageDetectionHandler(tabHelper: braveTranslateTabHelper)
     )
-    injectedScripts.append(braveTranslateScriptHandler)
+    injectedScripts.append(BraveTranslateScriptHandler(tabHelper: braveTranslateTabHelper))
```

Comment:
Please move the `BraveTranslateTabHelper` to be held by the `Tab` instead of the script handler, it shouldn't be owned by the script handler. Like I mentioned previously, the script handlers shouldn't be using the `Tab` unless they're handling JavaScript events.